### ISSUE TITLE
Change color of waypoints together with track

### ISSUE THREAD

**Issue by tradavyvy** [osmandapp/OsmAnd#12377]
# 🚀 feature request  or is it a bug?

### when changing the colour of an (imported) gpx-track it would be nice (and I would expect) if the embedded way-points would change to the same colour.

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/71978409/128300303-fa27898a-2502-4bcc-b0fa-44c91c4219ea.jpg"  width="300" height="auto"/> </td>
  </tr>
</table>

I like to import several (city) walks and then chose my own route by combining them on the map.

### Describe the solution you'd like

See above

### Describe alternatives you've considered

Live with it, or change all points manualy.

**Comment by tradavyvy** [comment]
And while you are at it, you might consider to use an other default marker for waypoints that come with a gps file than the circle star that is also the default for favorites. So it is clear which is which.

**Comment by sonora** [comment]
@vshcherb This issue has been dormant for some time.

But today it seems a gap to many users that our menu "Change appearance" for our "Tracks" provides all sorts of settings mostly related to the track line, but does not allow to change appearance of the waypoints?

Please note that increasingly, the GPX files we call "Tracks" are also used as exchange format by 3rd party web sites to mark just _groups of  'places'_ of some sort (with not necessarily any line in between). With this becoming a major use case, users increasingly expect to be able to modify the waypoint icons and color... would be a useful feature.

**Comment by vshcherb** [comment]
Yes it make sense now to start planning this feature

**Comment by sonora** [comment]
@vshcherb There probably needs to be an epic, at least a refactoring, unifying the appearance behavior and corresponding UI of
- for `<wpt>`: https://github.com/osmandapp/OsmAnd/issues/19971#issuecomment-2143732489
- for `<trk>`: https://github.com/osmandapp/OsmAnd/issues/19995#issuecomment-2144422839

to the effect that file-implicit (single or group) specs prevail unless the user specifically overrides them in OsmAnd's appearance settings.

**Comment by vshcherb** [comment]
@sonora Not sure , 1 issue is related to Dark Red so it's specific color, favorites still need to bereproduced. Not sure refactoring is needed.

I think user originally asked about changing color of waypoints which is possible to do for a group. So what seems more logical is to assign colors per Group and reuse colors and icons for Favorites / Tracks. Though in that case waypoints will not use Default Folder appearance.

**Comment by sonora** [comment]
@vshcherb  Yes, it can be done without code refactoring. Probably need a bit better a plan how we want it first, and then produce a separate issue or a small epic. Topics:
- Hierarchy in which appearance settings are honored (both for points and for track lines), I suggest file-internal spec is used (falling back to render default), unless the user specifically overrides.
- Should a user only be able to change the appearance for some/all _existing_ items, or also modify the appearance default (single / group) for items which would be imported in the future? (Changing the default sounds useful, but many users forget they have once manipulated this, then struggle with it later.)
- Unify the UI as much as possible: The track appearance UI should facilitate to manipulate the points, the track line, and the groups in the same fashion, perhaps on the same screen.
- The appearance dialog for favorites (both single ones and groups) should be similar, and follow the same look&feel like for GPX pont/track/group items.

Have not checked which of these points are already fully or partially in place. We definitely do not start at zero here, and this is not a big refactoring. But now everything should be done by one def, to introduce the consistency the historic development has not yet provided.

**Comment by vshcherb** [comment]
- Points appearance are much more complicated as it could have icons / shapes / color and it's also specific per group (see wikivoyage GPX files).
- User won't modify default appearance for folder as it's not trivial to find
- Most likely user will modify appearance per track which obviously needs to take priority over default (in line GPX settings)

Gpx files serve multiple purposes for different use cases as it's Data driven . By default importing track should display colors as intented inside files the only thing is somebody would apply default appearance for "Import" folder.

If users will change default appearance without acknowledge then we need to put warning or move to advanced stuff so they don't change without proper confirmation

**Comment by vshcherb** [comment]
Now we have dialog for GPX waypoint group to change default appearance, it works exactly the same as for favorites.
